deepchecks/nlp/checks/model_evaluation/weak_segments_performance.py

```diff
@@ -57,30 +58,35 @@ def run_logic(self, context: Context, dataset_kind) -> CheckResult:
 
         features, cat_features = get_relevant_data_table(text_data, data_type=self.segment_by,
                                                          columns=self.columns, ignore_columns=self.ignore_columns,
-                                                         n_top_features=self.n_top_features, add_label=True)
-
-        predictions = context.model.predict(text_data)
-
-        if self.loss_per_sample is not None:
-            loss_per_sample = self.loss_per_sample[text_data.get_original_text_indexes()]
-            proba_values = None
-        elif not hasattr(context.model, 'predict_proba'):
-            raise DeepchecksNotSupportedError('Predicted probabilities not supplied. The weak segment checks relies'
-                                              ' on log loss error that requires predicted probabilities, rather'
-                                              ' than only predicted classes.')
+                                                         n_top_features=self.n_top_features)
+
+        # Decide which scorer and score_per_sample to use in the algorithm run
+        encoded_dataset = self._target_encode_categorical_features_fill_na(features, text_data.label, cat_features)
+        if self.score_per_sample is not None:
+            score_per_sample = self.score_per_sample[list(features.index)]
+            scorer, dummy_model = None, None
+            avg_score = round(score_per_sample.mean())
```

Comment:
Why round? Why round here and not in display? Could lead to problems in very low ratio of un-annotated samples for e.g.

Reply:
you right, originally it was round to ,3